ui/src/components/ExportDialog.tsx

```diff
@@ -1,132 +1,246 @@
+    ddClient.desktopUI.dialog
+      .showOpenDialog({
+        properties: ["openDirectory"],
+      })
+      .then((result) => {
+        if (result.canceled) {
+          return;
+        }
 
-                setPath(result.filePaths[0]);
-            });
-    };
+        setPath(result.filePaths[0]);
+      });
+  };
 
-    const exportVolume = async () => {
-        setActionInProgress(true);
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFromRadioValue(
+      (event.target as HTMLInputElement).value as
+        | "directory"
+        | "local-image"
+        | "new-image"
+    );
+  };
 
-        console.log("volume name:", context.store.volume.volumeName);
-        console.log("path:", path);
-        console.log("fileName:", fileName);
+  const handleExport = async () => {
+    if (fromRadioValue === "directory") {
+      await exportVolume({ path, fileName });
+    } else if (fromRadioValue === "new-image") {
+      await exportToImage({ imageName: newImage });
+    } else {
+      await exportToImage({ imageName: image });
+    }
+    onClose(true);
+  };
 
-        // const encodedPath = encodeURIComponent(path)
-        // console.log("encodedPath:", encodedPath);
+  const renderDirectoryRadioButton = () => {
+    return (
+      <>
+        <FormControlLabel
+          value="directory"
+          control={<Radio />}
+          label="Local file"
+        />
+        {fromRadioValue === "directory" && (
+          <Stack pt={1} pb={2} pl={4}>
+            <Typography pb={1} variant="body2">
+              Create a compressed file (gzip’ed tarball) in a selected directory
+              with the content of a chosen volume.
+            </Typography>
 
-        ddClient.extension.vm.service
-            .get(`/volumes/${context.store.volume.volumeName}/export?path=${path}&fileName=${fileName}`)
-            .then((_: any) => {
-                ddClient.desktopUI.toast.success(
-                    `Volume ${context.store.volume.volumeName} exported to ${path}`
-                );
-            })
-            .catch((error) => {
-                ddClient.desktopUI.toast.error(
-                    `Failed to backup volume ${context.store.volume.volumeName} to ${path}: ${error.message}. HTTP status code: ${error.statusCode}`
-                );
-            })
-            .finally(() => {
-                setActionInProgress(false);
-                props.onClose();
-            })
-    };
+            <TextField
+              autoFocus
+              margin="dense"
+              id="file-name"
+              label="File name"
+              fullWidth
+              defaultValue={`${context.store.volume.volumeName}.tar.gz`}
+              spellCheck={false}
+              onChange={(e) => {
+                setFileName(e.target.value);
+              }}
+            />
+            <Grid container alignItems="center" gap={2}>
+              <Grid item flex={1}>
+                <TextField
```

Comment:
`TextField` is empty _after_ having selected the directory from the file picker. I'd like to see what directory has been selected.

Reply:
solved in [d0ea291](https://github.com/felipecruz91/vackup-docker-extension/pull/34/commits/d0ea291a948f729256acca5ef57c721c75d69c4c)

Reply:
Works like a charm now, thanks!